src/pages/ProfilePage/views/AccountSettingsView/AccountSettingsView.tsx

```diff
@@ -40,39 +59,51 @@ const AccountSettingsView = () => {
     return '';
   };
 
-  const setEditable = (field) =>
+  const setEditable = (field: Fields) =>
     setFormData((prevState) => ({ ...prevState, [`${field}Editable`]: !prevState[`${field}Editable`] }));
 
   const onChangeName = () => {
-    setEditable('name');
-    setFormData((prevState) => ({ ...prevState, namePrev: getValues('name') }));
+    setEditable(Fields.Name);
+    setFormData((prevState) => ({ ...prevState, namePrev: getValues(Fields.Name) as string }));
     // SEND REQUEST
   };
 
   const onChangePassword = () => {
-    setEditable('password');
-    setFormData((prevState) => ({ ...prevState, passwordPrev: getValues('password') }));
+    setEditable(Fields.Password);
+    setFormData((prevState) => ({ ...prevState, passwordPrev: getValues(Fields.Password) as string }));
     // SEND REQUEST
   };
 
-  const onCancelChange = (field) => {
+  const onCancelChange = (field: Fields) => {
     const fieldPrev = `${field}Prev`;
 
-    if (formData[fieldPrev]) {
-      setValue(field, formData[fieldPrev]);
+    if (formData[fieldPrev as keyof FormData]) {
+      setValue(field, formData[fieldPrev as keyof FormData] as never);
```

Comment:
This one's tricky. While `never` placates TS, it isn't the exact type here as there is a situation where there is a desired overlap of types between values of `FormData` and `ProfilePageSchemaInterface`. In the current state it is:
```
  const onCancelChange = (field: 'name' | 'password') => {
    const fieldPrev: 'namePrev' | 'passwordPrev' = `${field}Prev`;
    const prevValue = formData[fieldPrev]; // had to add this line because without this the type checker gets confused

    if (prevValue) {
      setValue(field, prevValue);
      setEditable(field);
    } else {
      setEditable(field);
    }
  };
```
However this way it is cumbersome and definitely could be refined. Please consider tweaking the types in this file to make the types of this functions work more less as it is above.

Reply:
I handled it in a different manner - tell me what you think about it.

Reply:
Well done with this!